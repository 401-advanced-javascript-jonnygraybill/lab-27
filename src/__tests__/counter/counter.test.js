import React from 'react';
// import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../../../src/components/app/app.js';
import Counter from '../../components/counter/counter.js';
import { exportAllDeclaration, isTSAnyKeyword } from '@babel/types';

describe('<Counter/>', () => {
  it('is alive at application start', () => {
    let app = shallow(<Counter />);
    expect(app.find('span').exists()).toBeTruthy();
  });

  it('decrements state on a click down', () => {
    let app = mount(<Counter />);
    let button = app.find('#decr');
    button.simulate('click');
    expect(app.state('count')).toBe(-1);
  });

  it('increments state on a click up', () => {
    let app = mount(<Counter />);
    let button = app.find('#incr');
    button.simulate('click');
    expect(app.state('count')).toBe(+1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});