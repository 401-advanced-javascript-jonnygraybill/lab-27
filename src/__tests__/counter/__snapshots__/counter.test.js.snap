// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly 1`] = `
<div
  id="counter"
>
  <button
    id="decr"
    onClick={[Function]}
  >
    -
  </button>
  <span>
    Count: 
    0
     
  </span>
  <button
    id="incr"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
